wagtail/admin/edit_handlers.py

```diff
@@ -789,19 +789,40 @@ def __init__(self, **kwargs):
 #Similarly, set up wagtailcore.Workflow to have edit handlers
 Workflow.panels = [
                     FieldPanel("name"),
-                    FieldPanel("active"),
                     InlinePanel("workflow_tasks", heading="Tasks"),
                     ]
 Task.panels = [
                     FieldPanel("name"),
-                    FieldPanel("active"),
                     ]
 GroupApprovalTask.panels = Task.panels + [FieldPanel('group')]
+# do not allow editing of group post creation - this could lead to confusing history if a group is changed after tasks
+# are started/completed
+GroupApprovalTask.exclude_on_edit = {'group'}
 
 Workflow.base_form_class = WagtailAdminModelForm
 Task.base_form_class = WagtailAdminModelForm
 
 
+class ExcludeFieldsOnEditMixin:
+    """A mixin for edit handlers, which disables fields listed in a model's 'exclude_on_edit' attribute when binding
+    to an existing instance - editing rather than creating"""
+
+    def bind_to(self, *args, **kwargs):
+        new = super(ExcludeFieldsOnEditMixin, self).bind_to(*args, **kwargs)
+        # when binding to an existing instance - ie editing - set those fields in the form to disabled
```

Comment:
I don't think we can assume that the presence of an instance means that we're editing - for example, the create page view also binds to a newly-created instance: https://github.com/wagtail/wagtail/blob/2e23ef4ee45ee553d0f1647a7a2ecb09983a19a3/wagtail/admin/views/pages.py#L216

(Perhaps checking `new.instance.pk is not None` would be sufficient to fix this?)

Reply:
Oh yes, good point - it worked for Tasks, but you're right that it wouldn't work elsewhere, thanks. I've added an additional check for `new.instance.pk is not None`